### PR TITLE
feat(dialog full screen): added reference to scrollable content

### LINKED ISSUE #3116: DialogFullScreen : Get a ref of the content
### Current behaviour
On the DialogFullScreen component, there is currently no way to get a ref of the content.

### Desired behaviour
I would like to be able to get a ref of the DialogFullScreen so I can know when a user scroll vertically in the dialog.

### Suggested solution
As there is a ref for the heading [headingRef](https://github.com/Sage/carbon/blob/f2563e14745942dabec9a17f096b8940ddb983e8/src/components/dialog-full-screen/dialog-full-screen.component.js#L29) is it possible to get a ref for the content ?

### Describe alternatives you've considered
I don't have any alternative at the moment.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 32.0.3
| react-scripts    | -
| React            | 16.13.1
| Browser          | Chrome
| npm              | 6.14.4
| Operating System | Windows 10

### PR DESCRIPTION
**Pull request by nicktitchmarsh**
fixes #3116, fixes FE-3021

### Proposed behaviour
Exposes a ref for the StyledContent element rendered inside a DialogFullScreen component

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`DialogFullScreen dialogTitle has a Ref to the scrollable content and the ref to content exists and contains html 1`] = `undefined`;
```
Comment (jamime) [inline src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap:3]: This snapshot should have content, not `undefined`.

File: src/components/dialog-full-screen/dialog-full-screen.spec.js [hunk 2]
```diff
@@ -136,6 +136,33 @@ describe('DialogFullScreen', () => {
...
+            title={ titleHeading }
+          >
+            <Button>Button</Button>
+            <Button>Button</Button>
+          </DialogFullScreen>
+        );
+      });
+
+      it('and the ref to content exists and contains html', () => {
+        expect(wrapper.instance().contentRef).toBeTruthy();
```
Comment (jamime) [inline src/components/dialog-full-screen/dialog-full-screen.spec.js:157]:
```suggestion
```

File: src/components/dialog-full-screen/dialog-full-screen.spec.js [hunk 3]
```diff
@@ -136,6 +136,33 @@ describe('DialogFullScreen', () => {
...
+        expect(wrapper.find(Button).length).toEqual(2);
```
Comment (jamime) [inline src/components/dialog-full-screen/dialog-full-screen.spec.js:158]:
```suggestion
```

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap (hunk 1 above)
Comment (nicktitchmarsh) [reply src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap]: Replaced with expected innerHTML

File: src/components/dialog-full-screen/dialog-full-screen.spec.js (hunk 2 above)
Comment (nicktitchmarsh) [reply src/components/dialog-full-screen/dialog-full-screen.spec.js]: removed

File: src/components/dialog-full-screen/dialog-full-screen.spec.js (hunk 3 above)
Comment (nicktitchmarsh) [reply src/components/dialog-full-screen/dialog-full-screen.spec.js]: removed

File: src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap (hunk 1 above)
Comment (jamime) [reply src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap]: The unit tests are failing at the moment for this
Did the test generate this for you or did you add it manually?

File: src/components/dialog-full-screen/dialog-full-screen.spec.js [hunk 4]
```diff
@@ -136,6 +136,31 @@ describe('DialogFullScreen', () => {
...
+            title={ titleHeading }
+          >
+            <Button>Button</Button>
+            <Button>Button</Button>
+          </DialogFullScreen>
+        );
+      });
+
+      it('and the ref to content exists and contains html', () => {
+        expect(wrapper.instance().contentRef.current.innerHtml).toMatchSnapshot();
```
Comment (ianforster-sage) [inline src/components/dialog-full-screen/dialog-full-screen.spec.js:157]: I think your snapshot needs to be refreshed. This test is currently failing.

**Review by ianforster-sage — CHANGES_REQUESTED** [review]

File: src/components/dialog-full-screen/dialog-full-screen.spec.js (hunk 4 above)
Comment (nicktitchmarsh) [reply src/components/dialog-full-screen/dialog-full-screen.spec.js]: Updated can now be reviewed

File: src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap (hunk 1 above)
Comment (nicktitchmarsh) [reply src/components/dialog-full-screen/__snapshots__/dialog-full-screen.spec.js.snap]: Updated again to JSON

**Review by jamime — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Pull request merged**